pkg/availability/multisigcollector/parts/certcreation/certcreation.go

```diff
@@ -95,17 +94,18 @@ func IncludeCreatingCertificates(
 	// When the mempool provides a batch, compute its ID.
 	mempooldsl.UponNewBatch(m, func(txIDs []tt.TxID, txs []*requestpbtypes.Request, context *requestBatchFromMempoolContext) error {
 		if len(txs) == 0 {
+			// If the batch is empty, immediately stop it and respond to all pending requests with an empty certificate.
 			delete(state.Certificates, context.reqID)
-			if len(state.RequestStates) > 0 { // do not do work for the empty batch if there is not even a request to answer
-				sendIfReady(m, &state, params, true)
-				if len(state.Certificates) == 0 {
-					apbdsl.ComputeCert(m, mc.Self)
-				}
-			}
-			return nil
+			sendEmptyCert(m, &state)
```

Comment:
This seems to send an empty certificate when receiving a batch, while `sendIfReady` responds with all the so-far delivered batches if any available, and only with an empty cert if none are available. I may be missing something perhaps thinking

Reply:
As far as I understand, the situation where there are requests pending **and** certificates available is impossible. If it is possible, that's a bug, because it definitely shouldn't be - those requests should have been responded to with the existing certificates as soon as that situation occurred. Thus, if there is an empty certificate to be sent, it will certainly be the only one.

Reply:
Good point, makes sense